File: src/main/java/com/github/sorusclient/client/adapter/RenderBuffer.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter;

import com.google.common.collect.ImmutableList;

import java.util.ArrayList;
import java.util.List;

public class RenderBuffer {

    private DrawMode drawMode;
    private final List<Vertex> vertices = new ArrayList<>();

    public void push(Vertex vertex) {
        vertices.add(vertex);
    }

    public void setDrawMode(DrawMode drawMode) {
        this.drawMode = drawMode;
    }

    public DrawMode getDrawMode() {
        return drawMode;
    }

    public List<Vertex> getVertices() {
        return ImmutableList.copyOf(vertices);
    }

    public enum DrawMode {
        QUAD
    }

}

File: src/main/kotlin/com/github/sorusclient/client/adapter/RenderBuffer.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter

class RenderBuffer {
    var drawMode: DrawMode? = null
    val vertices: MutableList<Vertex> = ArrayList()

    fun push(vertex: Vertex): RenderBuffer {
        vertices.add(vertex)
        return this
    }

    enum class DrawMode {
        QUAD
    }
}